src/SPC/builder/traits/UnixGoCheckTrait.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SPC\builder\traits;
+
+use SPC\doctor\CheckResult;
+use SPC\exception\RuntimeException;
+use SPC\store\Downloader;
+use SPC\store\FileSystem;
+
+trait UnixGoCheckTrait
+{
+    private function checkGoAndXcaddy(): ?CheckResult
+    {
+        $paths = explode(PATH_SEPARATOR, getenv('PATH'));
+        $goroot = getenv('GOROOT') ?: '/usr/local/go';
+        $goBin = "{$goroot}/bin";
+        $paths[] = $goBin;
+        if ($this->findCommand('go', $paths) === null) {
+            $this->installGo();
+        }
+
+        $gobin = getenv('GOBIN') ?: (getenv('HOME') . '/go/bin');
+        putenv("GOBIN={$gobin}");
+
+        $paths[] = $gobin;
+
+        if ($this->findCommand('xcaddy', $paths) === null) {
+            shell(true)->exec('go install github.com/caddyserver/xcaddy/cmd/xcaddy@latest');
+        }
+
+        return CheckResult::ok();
```

Comment:
the go install locations are what I'm least satisfied with. I don't like leaving a `go` folder under $HOME, but also can't think of a better way